x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/CannedSourceOperator.java

```diff
@@ -96,5 +96,9 @@ public Page getOutput() {
     }
 
     @Override
-    public void close() {}
+    public void close() {
+        while (page.hasNext()) {
+            page.next().releaseBlocks();
+        }
```

Comment:
This let's me assert that we've closed all the input pages even if these's an error!